crates/sncast/tests/integration/data_transformer.rs

```diff
@@ -0,0 +1,1022 @@
+        contract_class,
+        &get_selector_from_name("signed_fn").unwrap(),
+    )?;
+
+    let expected_output = vec![Felt::from(-273i16)];
+
+    assert_eq!(result, expected_output);
+
+    Ok(())
+}
+
+// Problem: Although transformer fails to process the given input as `i32`, itthen succeeds to interpret it as `felt252`
+// Overflow checks will not work for functions having the same serialized and Cairo-like calldata length.
+// User must provide a type suffix or get the invoke-time error
+#[ignore = "Impossible to pass with the current solution"]
+#[tokio::test]
+async fn test_signed_fn_overflow() {
+    let simulated_cli_input = vec![(i32::MAX as u64 + 1).to_string()];
+
+    let contract_class = CLASS.get_or_init(init_class).await.to_owned();
+    let selector = get_selector_from_name("signed_fn").unwrap();
+
+    let output = transform(&simulated_cli_input, contract_class, &selector);
+
+    output
+        .unwrap_err()
+        .assert_contains(r#"Failed to parse value "2147483648" into type "i32""#);
+}
+
+#[tokio::test]
+async fn test_signed_fn_overflow_with_type_suffix() {
+    let simulated_cli_input = vec![format!("{}_i32", i32::MAX as u64 + 1)];
+
+    let contract_class = CLASS.get_or_init(init_class).await.to_owned();
+    let selector = get_selector_from_name("signed_fn").unwrap();
+
+    let result = transform(&simulated_cli_input, contract_class, &selector);
+
+    result
+        .unwrap_err()
+        .assert_contains(r#"Failed to parse value "2147483648" into type "i32""#);
+}
+
+#[tokio::test]
+async fn test_happy_case_unsigned_function_cairo_expressions_input() -> anyhow::Result<()> {
+    let contract_class = CLASS.get_or_init(init_class).await.to_owned();
+
+    let input = vec![u32::MAX.to_string()];
+
+    let result = transform(
+        &input,
+        contract_class,
+        &get_selector_from_name("unsigned_fn").unwrap(),
+    )?;
+
+    let expected_output = vec![Felt::from(u32::MAX)];
+
+    assert_eq!(result, expected_output);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_happy_case_unsigned_function_serialized_input() -> anyhow::Result<()> {
+    let contract_class = CLASS.get_or_init(init_class).await.to_owned();
+
+    let input = vec![Felt::from(u32::MAX).to_hex_string()];
+
+    let result = transform(
+        &input,
+        contract_class,
+        &get_selector_from_name("unsigned_fn").unwrap(),
+    )?;
+
+    let expected_output = vec![Felt::from(u32::MAX)];
+
+    assert_eq!(result, expected_output);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_happy_case_tuple_function_cairo_expression_input() -> anyhow::Result<()> {
+    let contract_class = CLASS.get_or_init(init_class).await.to_owned();
+
+    let input = vec![String::from("(2137_felt252, 1_u8, Enum::One)")];
+
+    let result = transform(
+        &input,
+        contract_class,
+        &get_selector_from_name("tuple_fn").unwrap(),
+    )?;
+
+    let expected_output: Vec<Felt> = vec![2137.into(), 1.into(), 0.into()];
```

Comment:
Why does it have `0` as the last value? Shouldn't enum with no storage be serialized just as `1`?

Reply:
This is enum's first variant serialization (0 from here because of first, we count from 0) and there is no value so nothing more